ckan/templates/package/snippets/resource_form.html

```diff
@@ -72,16 +72,16 @@
       {% block previous_button %}
         <button class="btn btn-default" name="save" value="go-dataset" type="submit">{{ _('Previous') }}</button>
       {% endblock %}
-      {% block again_button %}
-        <button class="btn btn-default" name="save" value="again" type="submit">{{ _('Save & add another') }}</button>
-        {% endblock %}
-      {% block save_button %}
-      <button class="btn btn-primary" name="save" value="go-metadata" type="submit">{% block save_button_text %}{{ _('Finish') }}{% endblock %}</button>
-      {% endblock %}
-    {% else %}
-      {% block add_button %}
-      <button class="btn btn-primary" name="save" value="go-dataset-complete" type="submit">{{ _('Add') }}</button>
-      {% endblock %}
     {% endif %}
+    {% block again_button %}
+      <button class="btn btn-default" name="save" value="again" type="submit">{{ _('Save & add another') }}</button>
+    {% endblock %}
+    {% block save_button %}
+      {% if stage %}
+        <button class="btn btn-primary" name="save" value="go-metadata" type="submit">{{ _('Finish') }}</button>
+      {% else %}
+        <button class="btn btn-primary" name="save" value="go-dataset-complete" type="submit">{{ _('Finish') }}</button>
```

Comment:
Can we keep the "Add" label on this case? I think it makes more sense to keep the existing message.